src/Microsoft.OData.Core/ODataInnerError.cs

```diff
@@ -93,5 +136,39 @@ internal string ToJson()
                 this.StackTrace == null ? "" : JsonValueUtils.GetEscapedJsonString(this.StackTrace),
                 this.InnerError == null ? "{}" : this.InnerError.ToJson());
         }
-}
+
+        private string GetStringValue(string propertyKey)
+        {
+            if (Properties.ContainsKey(propertyKey))
+            {
+                return Properties[propertyKey].FromODataValue()?.ToString();
+            }
+
+            return string.Empty;
+        }
+
+        private void SetStringValue(string propertyKey, string value)
+        {
+            if (!Properties.ContainsKey(propertyKey))
+            {
+                if (value == null)
+                {
+                    Properties.Add(propertyKey, new ODataNullValue());
+                    return;
+                }
+
+                Properties.Add(propertyKey, value.ToODataValue());
+                return;
+            }
+
+            if (value == null)
+            {
+                Properties[propertyKey] = new ODataNullValue();
+                return;
+            }
+
+            Properties[propertyKey] = value.ToODataValue();
+            return;

```

Comment:
>return [](start = 12, length = 6)

unneeded return. in general, this code follows the "early return pattern". I don't have a strong bias either way, but most of our code (especially for void methods) has a single (or few, especially success) exit points.  Not sure if there's a preferred style across Microsoft. We should discuss what coding pattern we want to follow.